### ISSUE TITLE
Option to use as display-only (read-only mode)

### ISSUE THREAD

**Issue by tonio-ramirez** [willowsystems/jSignature#9]
Would be nice to have an option to disable user entry and use the plugin for displaying signatures only, without having to generate an image and use that in place of the signature area.

**Comment by dvdotsenko** [comment]
It turns out I need that option too. Due to the lucky coincidence, I am also a guy who can also do the change and commit. You'll see it up today / tomorrow.

Daniel

**Comment by tonio-ramirez** [comment]
That is great news! :D

**Comment by RodEsp** [comment]
Any updates on this?
I don't see anything in the documentation or in the code of jSignature.js that would allow us to set the canvas as read-only.

**Comment by dvdotsenko** [comment]
The internal need for that feature had disappeared. 

The definition of the feature is also unclear. Example:
- Does "read-only" mean you forbid to "clear" it too?
- How does it become "read-only"? Init-time? A switch of some sort that locks a signature in place?
- What API needs to be disabled along with it? 
- Because of the various combinations of the definitions, do we need to invent new option keys for switching between them?

Because of the lack of need and ambiguity of the API this is not on a priority list right now.

**Comment by tonio-ramirez** [comment]
Well, I can't do much about your internal need disappearing, but I can clarify the definition of the feature.

By "read-only", I mean the signature data (and image) can't be affected by user input (i.e. the panel/canvas does not respond to mouse events by drawing lines or changing the vector data).  The data can be set programmatically via normal API calls.  So yes, it would forbid "clear" as well.

How it becomes "read-only"... well, for my purposes, it can be at init-time, but it would be more versatile if it could be, like, you mention, a switch of some sort that locks a signature in place.

As to what API needs to be disabled, I would suggest that either no part of the API be disabled (allowing full control of the data + image via the API, but none through user input [mouse events]), or alternatively all parts of the API that deal with changing the data and/or image (which would make the signature truly read-only, unable to be changed even programmatically).  I prefer the former option (no response to mouse events, but full API access), but am fine with either.

Inventing new option keys would allow for complex combinations (allow clear, but not drawing; allow modifying API calls, but not mouse events; disallow all modifications, via API or mouse events; etc.), but in my opinion such complexity is unwarranted.  The most useful part of this feature is, I belive, the ability to display signatures without allowing a user to change them.  This can be acheived by a switch that enables and disables the currently normal responses to mouse events; or, alternatively, by an init-time option that disables said responses.

**Comment by dvdotsenko** [comment]
It looks like the buttons (Undo button and the "Sign Here" sticker) need to be disabled too)

**Comment by mc2** [comment]
+1 here.

Client wants signature areas to always 'look the same' on the form; so images used only in printing.

Just having init-time 'readOnly: true' works for me. I can 'UndoButton': false at init-time, not display custom 'Clear' buttons.

The "Sign Here" sticker doesn't show up when there's data, but it would be nice to have blank 'readOnly' without it.

I have use cases where a form is signed, but needs 
1) further (authorized) signatures in workflow, or 
2) displayed with signatures to users who can't make changes.

Thanks!

**Comment by webdev36** [comment]
Hello

Glad to use your jSignature.
But I really want to make disable feature, as I should show the stored signature image to the users who can't edit signature anymore.
It is same as locking jSignature canvas.
It would be great if you can help me.

Thanks!

**Comment by acsipak** [comment]
+1

**Comment by bvisonl** [comment]
+1

**Comment by TheKuddlyOne** [comment]
+1

Additionally, it looks as though brinley made a readOnly parameter in https://github.com/brinley/jSignature/commit/0fe620bb3d7a930795f2b2e7a2f25bd861311718 but it never seemed to have been pushed?

**Comment by bkdotcom** [comment]
easy enough...

just add a transparent div above the signature
`<div style="position:absolute; top:0; bottom:0; left:0; right: 0;"></div>`

**Comment by AP-TGR** [comment]
I agree to @webmaster219 , we need to display the signature to the user who does not have access to edit it. If you can't make it disabled provide some alternate option to display the signature in read only mode.
